app/globals.py

```diff
@@ -18,6 +19,10 @@ def get_questionnaire_store(user_id, user_ik):
         storage = EncryptedQuestionnaireStorage(user_id, user_ik, pepper)
         store = g._questionnaire_store = QuestionnaireStore(storage)
 
+    if store.metadata:  # When creating the questionnaire storage, there is no metadata for upgrading
```

Comment:
I don't understand what this is for?

Reply:
When this is called: https://github.com/ONSdigital/eq-survey-runner/blob/7846f721c89ec3aa0cc27c4544188c1796b3539f/app/authentication/authenticator.py#L109 it appeared that metadata had not been populated, so it was unable to fetch the schema

Reply:
Maybe `schema = load_schema_from_metadata(store.metadata)` should be within the if at 
https://github.com/ONSdigital/eq-survey-runner/pull/1727/files/283be2434be900ecbaeea3e758300e5f815f6a8b..9337df97fea6b7c6dc530ec52280caaf2618732e#diff-dc061d8bc41d7eabbb038d35ad5ae467R32

otherwise we are loading the schema when we don't need it.

Reply:
I figured that it's only fetching a reference to the cached schema? I didn't want to put the logic for fetching the schema inside the QuestionnaireStore unit particuarly